src/js/profiling/hermes.ts

```diff
@@ -0,0 +1,91 @@
+import { NATIVE } from '../wrapper';
+import { convertToSentryProfile } from './convertHermesProfile';
+import type { RawThreadCpuProfile } from './types';
+
+export type StackFrameId = number;
+export type MicrosecondsSinceBoot = string;
+
+export interface TraceEvent {
+  name: string;
+  ph: string;
+  cat: string;
+  pid: number;
+  ts: MicrosecondsSinceBoot;
+  tid: string;
+  args: {
+    name?: string;
+  };
+}
+
+export interface Sample {
+  cpu: string;
+  name: string;
+  ts: MicrosecondsSinceBoot;
+  pid: number;
+  tid: string;
+  weight: string;
+  sf: StackFrameId;
+}
+
+export interface StackFrame {
+  line?: string;
+  column?: string;
+  funcLine?: string;
+  funcColumn?: string;
+  name: string;
+  category: string;
+  parent?: number;
+}
+
+export interface Profile {
+  traceEvents: TraceEvent[];
+  samples: Sample[];
+  stackFrames: Record<string, StackFrame>;
+}
+
+export interface HermesStackFrameNameParsed {
+  function?: string;
+  fileName?: string;
+}
+
+const FILE_NAME_REGEX = /([a-zA-Z0-9_-]+\.[a-zA-Z0-9_-]+)(?::(\d+:\d+))?\)/;
+/**
+ * Hermes Profile Stack Frame Name contains function name and file path.
+ *
+ * `foo(/path/to/file.js:1:2)` -> `{ function: 'foo', fileName: 'file.js' }`
```

Comment:
does line and column info correspond to call site or function definition?

Reply:
call site - but we only parse the func name and the path

Can you share a bit more context, could it be a trouble somewhere?

Reply:
No, just something worth knowing tbqh. Some profilers can output line/col of where a fn is defined - this can have an impact on the interpretation of the charts/graphs as well as reduce payload size.